modin/pandas/test/test_series.py

```diff
@@ -4258,11 +4259,33 @@ def test_encode(data, encoding_type):
 
 
 @pytest.mark.parametrize("data", test_string_data_values, ids=test_string_data_keys)
-def test_add_string_to_series(data):
+def test_non_commutative_add_string_to_series(data):
+    # This test checks that add and radd do different things when addition is
+    # not commutative, e.g. for adding a string to a string. For context see
+    # https://github.com/modin-project/modin/issues/4908
     eval_general(*create_test_series(data), lambda s: "string" + s)
     eval_general(*create_test_series(data), lambda s: s + "string")
 
 
+def test_non_commutative_multiply_pandas():
+    # The non commutative integer class implementation is tricky. Check that
+    # multiplying such an integer with a pandas series is really not
+    # commutative.
+    pandas_series = pd.DataFrame([[1]], dtype=int)
+    integer = NonCommutativeMultiplyInteger(2)
+    assert not (integer * pandas_series).equals(pandas_series * integer)
```

Comment:
I wonder if we really need two tests testing that non-commutative int implementation... though I do understand that it might behave differently in pandas for DF or Series

Reply:
I prefer the extra security of the two tests.